### DISCUSSION TITLE
Budibase backup/restore procedure with cli fails or procedure clarifications are needed.

### DISCUSSION THREAD

**Discussion by gmentzik** [Budibase/budibase#8270] [Help]
On a linux mint ( ubuntu ) I am trying to do a backup restore simulation.
1) Prepare to Install a clean budibase via docker compose after I have downloaded docker-compose and .env via budi hosting --init.
2) To docker compose file I add an admin user as well as a mysql service to use from my app.
3) docker-compose up or budi hosting --start, (tried both), and login as admin. Create an app reading tables from mysql (just auto pages). Also added a new table in apps internal db with an image attachment (to test file backup/restore).
4) Log off 
5) With budubase running I perform a backup with budi cli using the .env file.
6) docker-compose down. Delete the 3 budibase docker volumes.
7) Start budibase containers (docker-compose up -d or budi hosting --start). 
8) Check that budibase is running by hitting localhost:10000 . (Tried scenario to login as admin or not before restore).
9) Use Budibase cli to restore backup. 

Result:  Depending if I had logged in before restoring or not I get various errors. Either I cannot loggin or if I manage to loggin the app is in the list but you cannot open it. I have tried both the Linux cli app and the nodejs npm installation. 

Since at documentation the backup and restore procedure with cli is not well described, could you please tell me what are the correct backup and restore steps? 

I attach images from errors I am getting after restore:
1. Not having logged in before restore:
![Screenshot from 2022-10-15 19-30-35](https://user-images.githubusercontent.com/65217733/196022699-449a08bf-8e2b-42c5-a018-3211757eaeb4.png)
2. Having logged in before restore and trying to click on my app to edit.
![Screenshot from 2022-10-15 12-06-14](https://user-images.githubusercontent.com/65217733/196022708-21e54f0a-1a1a-442d-b410-6f41f08cd88a.png)

**Comment by sahjanivishal** [comment]
Facing the same error as this

**Comment by sahjanivishal** [comment]
@shogunpurple @melohagan Can you please help with this issue?

**Reply by melohagan** [reply]
Hey @gmentzik @sahjanivishal To verify, you did follow through the steps outlined here? https://docs.budibase.com/docs/backups

**Reply by gmentzik** [reply]
I read the documentation and did the steps I describe above.

**Comment by sahjanivishal** [comment]
Hey @melohogan

Yes I followed the same steps mentioned and was able to import the backup.
I am also able to create the new apps but the existing app in the backup
shows the error message "Error initialising app:"

**Reply by melohagan** [reply]
This issue may be related: https://github.com/Budibase/budibase/issues/8401

**Reply by sahjanivishal** [reply]
This one has a bug label. Will it be possible to try recreating the issue on your side? This could be impacting a large number of users but since not many people test their backups there are less issues reported

**Reply by gmentzik** [reply]
I have another question for backup. At backup page you say that if we backup the docker volume folder is also a valid backup restore procedure. I have tested this and successfully managed to restore my budibase system to another host machine. But to do that I also backup up and restored the docker images used at the original installation. 
So my questions are:
 **Is it necessary to backup the docker images to have the same versions used when backup was taken, or the data folders are irrelevant of the budibase, couch_db3, minio and redis images versions?
Can I restore docker volumes to Budibase host created with latest docker-compose even if backup was taken with an older version?**

**Reply by gmentzik** [reply]
Also if when you try to recreate the backup/restore scenario with budibase cli tool, could you please record a video showing the steps. I think it would be much helpful for everyone!

**Reply by mike12345567** [reply]
Hi @gmentzik and @sahjanivishal - i've had a look into this - I believe there might be a few issues with our CLI + our proxy image in different configurations, I've got some fixes together which will go out in our next release (should be early next week).

**Reply by sahjanivishal** [reply]
Hi @mike12345567 , any update if these fixes are included in the recent versions? I tried again with the 2.1.9 version but still getting the same error.

**Reply by mike12345567** [reply]
Hi @sahjanivishal - this is functioning in the latest version of the CLI. You will need to update your CLI - make sure that `budi --version` shows a version greater than `2.1.13`.

**Comment by sahjanivishal** [comment]
Hello @mike12345567 , thank you for shipping this quickly! I tried updating the budibase cli and I'm on version 2.1.15 now. But now when I try to run budi backups --export --env .env, I get the below error

Error - Error: Unable to access MinIO/S3 - check environment config.

I have not changed any of the content of the docker compose file or the yaml file.

**Reply by melohagan** [reply]
Hey @sahjanivishal 

Maybe you're just missing some env variables setup: https://docs.budibase.com/docs/backups#setting-up

**Reply by sahjanivishal** [reply]
hi @melohagan , I have defined the required env variables in my .env file and I was able to export backups prior to the latest update.

**Reply by giacomouberti** [reply]
Hi there, any updates on this issue? I'm facing the same problem and double checked all the variables and followed instructions from docs but I keep getting the same error: Error - Error: Unable to access MinIO/S3 - check environment config.

It's also not clear to me why MINIO_URL should be MINIO_URL=http://localhost:10000

I have tryed also exposing minio as descriped here https://docs.budibase.com/docs/accessing-minio but and update ports but it's not solving the problem
